internal/services/tenant_isolation_policy/resource_tenant_isolation_policy.go

```diff
@@ -0,0 +1,562 @@
+	updatedState.Timeouts = timeoutsVal
+
+	// Set the state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &updatedState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Updated tenant isolation policy with ID %s", updatedState.Id.ValueString()))
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *Resource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	ctx, exitContext := helpers.EnterRequestContext(ctx, r.TypeInfo, req)
+	defer exitContext()
+
+	var state TenantIsolationPolicyResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tenantId := state.Id.ValueString()
+	if tenantId == "" {
+		resp.Diagnostics.AddError(
+			"Missing tenant ID",
+			"The tenant ID is unexpectedly missing from state. This is likely a provider bug.",
+		)
+		return
+	}
+
+	// To delete the policy, we update it with an empty policy
+	emptyPolicy := TenantIsolationPolicyDto{
+		Properties: TenantIsolationPolicyPropertiesDto{
+			TenantId:       tenantId,
+			IsDisabled:     types.BoolValue(false).ValueBoolPointer(),
+			AllowedTenants: []AllowedTenantDto{},
+		},
+	}
+
+	_, err := r.Client.createOrUpdateTenantIsolationPolicy(ctx, tenantId, emptyPolicy)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting tenant isolation policy",
+			fmt.Sprintf("Could not delete tenant isolation policy: %s", err.Error()),
+		)
+		return
+	}
+
+	// Remove the resource from state
+	resp.State.RemoveResource(ctx)
+
+	tflog.Info(ctx, fmt.Sprintf("Deleted tenant isolation policy with ID %s", tenantId))
+}
+
+// ImportState imports the resource into Terraform from an existing tenant isolation policy.
+func (r *Resource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	ctx, exitContext := helpers.EnterRequestContext(ctx, r.TypeInfo, req)
+	defer exitContext()
+
+	// The import ID can either be empty (use current tenant) or a specific tenant ID
+	tenantId := req.ID
+	if tenantId == "" {
+		// Get the current tenant ID if not provided
+		tenantInfo, err := r.Client.TenantApi.GetTenant(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error retrieving tenant information",
+				fmt.Sprintf("Could not retrieve tenant information: %s", err.Error()),
+			)
+			return
+		}
+		tenantId = tenantInfo.TenantId
+	}
+
+	// Set the ID into the state
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), tenantId)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Imported tenant isolation policy with ID %s", tenantId))
+}
+
+// Helper functions
+
+// convertToDto converts the Terraform model to the API DTO.
+func convertToDto(ctx context.Context, tenantId string, model *TenantIsolationPolicyResourceModel) (*TenantIsolationPolicyDto, diag.Diagnostics) {
```

Comment:
Move the helper function that convert DTO to model and model to Dto to models.go file

Reply:
👍